### PR TITLE
Optimize Concat operation

### PR DESCRIPTION
**Pull request by mbencer**
- removed slow `CoordinateTransform` usage
- reduced memory footprint by not using templates

### CODE REVIEW

**Comment by jane-intel** [comment]
Great. Please add some tests.

**Comment by mbencer** [comment]
> Great. Please add some tests.

Added a few edge tests to constant folding

File: ngraph/core/include/ngraph/runtime/reference/concat.hpp [hunk 1]
```diff
@@ -16,68 +16,22 @@
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/runtime/reference/concat.hpp:16]: Please move your reference implementation to the [library with reference implementations](https://github.com/openvinotoolkit/openvino/tree/master/ngraph/core/reference).

Comment (mbencer) [reply ngraph/core/include/ngraph/runtime/reference/concat.hpp]: done

**Comment by jane-intel** [comment]
Please merge the latest master

**Pull request merged**